codegen/src/codegen.ts

```diff
@@ -134,15 +113,18 @@ export async function generate({
       XataRecord
     } from '@xata.io/client';
 
+    type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
+    type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>
+
     ${tables.map((table) => generateTableType(table)).join('\n')}
 
     const links = ${JSON.stringify(links)};
 
     export class XataClient extends BaseClient<{
       ${tables.map((table) => `"${table.name}": Repository<${getTypeName(table.name)}>;`).join('\n')}
     }> {
-      constructor(options: XataClientOptions) {
-        super(options, links);
+      constructor(options: ${config.dbName ? `PartialBy<XataClientOptions, 'databaseURL'>` : 'XataClientOptions'}) {
```

Comment:
I thought `config.dbName` would be always defined. Is that not the case?

Reply:
Great question. I'm thinking ahead a bit here for cases where the Xata CLI (outside of our control) changes and `dbName` may not be the same key or similar. Happy to remove this if you think it's simpler. Let me know.

Reply:
🤔  I'm not sure if I follow. As far as I see in the code, if the key changes, the zod validation would fail, right? (I'm not super familiar with zod, but looks like you'd have to use `z.string().optional()` if you want to cover that case).

See https://github.com/xataio/client-ts/pull/44/files#diff-74e5a79804f41cdb60666c9eab8560b0f6ab27bf8f1936b5058e3a90f36e03bbR5

Anyway I don't think we should cover that case. It is very unlikely that the format will change in that way, and in that case the validation during parsing should fail.

In order to prevent backwards-incompatible changes in configuration files, I think the best practice is using a version number inside the file itself (such as `lockfileVersion` in `package-lock.json`).

Reply:
Cool, done in https://github.com/xataio/client-ts/pull/44/commits/5bef124beaaa4d3d4cd0c76b7f43a957841dd556.